### PR TITLE
check na data at init_training

### PR DESCRIPTION
**Pull request by Jarvi-Izana**
#2785 #2784

early termination of a training job if data contains na. 

Also, this pr reorganized `init_training` a little bit.

### CODE REVIEW

File: src/python/turicreate/test/test_drawing_classifier.py [hunk 1]
```diff
@@ -96,8 +97,20 @@ def setUpClass(self, warm_start=None):
...
+        try:
+            _tc.drawing_classifier.create(sf, self.target)
+        except _ToolkitError as e:
+            self.assertTrue("dropna" in str(e))
+
+    def test_create_with_missing_value_in_label(self):
+        sf = self.check_cross_sf
+        sf = sf.remove_column(self.target)
+        sf = sf.add_column(_tc.SArray(
+            [None for _ in range(len(sf))], dtype=str), self.target)
```
Comment (TobyRoseman) [inline src/python/turicreate/test/test_drawing_classifier.py]: Rather than `[None for _ in range(len(sf))]` you can just do `[None] * len(sf)`.

Comment (Jarvi-Izana) [reply src/python/turicreate/test/test_drawing_classifier.py]: Wow, will try.

File: src/python/turicreate/test/test_drawing_classifier.py [hunk 2]
```diff
@@ -96,8 +97,20 @@ def setUpClass(self, warm_start=None):
...
+        try:
             _tc.drawing_classifier.create(sf, self.target)
+        except _ToolkitError as e:
+            self.assertTrue("dropna" in str(e))
```
Comment (jakesabathia2) [inline src/python/turicreate/test/test_drawing_classifier.py]: good!

**Comment by Jarvi-Izana** [comment]
passed the internal pipeline. merge it.

**Pull request merged**